WordPressAuthenticatorTests/Authenticator/WordPressAuthenticatorTests.swift

```diff
@@ -38,21 +45,352 @@ class WordPressAuthenticatorTests: XCTestCase {
         loginFields.username = email
         WordPressAuthenticator.storeLoginInfoForTokenAuth(loginFields)
-
+        
         var retrievedLoginFields = WordPressAuthenticator.retrieveLoginInfoForTokenAuth()
         var retrievedEmail = retrievedLoginFields.username
         XCTAssert(email == retrievedEmail, "The email retrived should match the email that was saved.")
-
+        
         WordPressAuthenticator.deleteLoginInfoForTokenAuth()
         retrievedLoginFields = WordPressAuthenticator.retrieveLoginInfoForTokenAuth()
         retrievedEmail = retrievedLoginFields.username
-
+        
         XCTAssert(email != retrievedEmail, "Saved loginFields should be deleted after calling deleteLoginInfoForTokenAuth.")
     }
+    
+    func testAuthenticatorInitHasCorrectProperties() {
+        //COME BACK TO THIS
+        let authenticator = WordPressAuthenticator.shared
+        
+        XCTAssertEqual(authenticator.configuration.wpcomClientId, "23456")
+    }
+    
+    func testSupportPushNotificationReceived() {
+        let _ = expectation(forNotification: .wordpressSupportNotificationReceived, object: nil, handler: nil)
+        
+        WordPressAuthenticator.shared.supportPushNotificationReceived()
+        
+        waitForExpectations(timeout: timeInterval, handler: nil)
+    }
+    
+    func testSupportPushNotificationCleared() {
+        let _ = expectation(forNotification: .wordpressSupportNotificationCleared, object: nil, handler: nil)
+        
+        WordPressAuthenticator.shared.supportPushNotificationCleared()
+        
+        waitForExpectations(timeout: timeInterval, handler: nil)
+    }
+    
+    func testWordpressAuthIsAuthenticationViewController() {
+        let loginViewcontroller = LoginViewController()
+        let nuxViewController = NUXViewController()
+        
+        XCTAssertTrue(WordPressAuthenticator.isAuthenticationViewController(loginViewcontroller))
+        XCTAssertTrue(WordPressAuthenticator.isAuthenticationViewController(nuxViewController))
+    }
+    
+    func testIsGoogleAuthURL() {
+        let url = URL(string: "https://google.com")!
+        
+        XCTAssertTrue(WordPressAuthenticator.shared.isGoogleAuthUrl(url))
+    }
+    
+    func testIsWordPressAuthURL() {
+        let url = URL(string: "https://magic-login")!
+        
+        XCTAssertTrue(WordPressAuthenticator.shared.isWordPressAuthUrl(url))
+    }
+    
+    func testHandleWordPressAuthURLReturnsTrueOnSucceed() {
+        let url = URL(string: "https://wordpress.com/wp-login.php?token=1234567890%26action&magic-login&sr=1&signature=1234567890oienhdtsra")
+        
+        XCTAssertTrue(WordPressAuthenticator.shared.handleWordPressAuthUrl(url!, allowWordPressComAuth: true, rootViewController: UIViewController()))
+    }
+    
+    func testSignInForWPOrgReturnsVC() {
+        let vc = WordPressAuthenticator.signinForWPOrg()
+        
+        XCTAssertTrue((vc as Any) is LoginSiteAddressViewController)
```

Comment:
Just curious, did this not work?

```swift
XCTAssertTrue(vc is LoginSiteAddressViewController)
```

Reply:
I tested this a bit and funny enough it depends on the complexity of the function being tested.  For functions like:
`public class func signinForWPCom() -> LoginEmailViewController`
that function doesn't do much except create the VC and return it (no modifications to the controller at all) if you ask vc is LoginEmailViewController that will always succeed because that is what the function returns.  So that kinda makes that test not very interesting, it is just confirming the function actually works, but to do so first you have to mark it as any or the `is` will always succeed

But the other functions like `signinForWPOrg() -> UIViewController` defines a LoginSiteAddressViewController and then returns it as a UIViewController, and in that case you don't need to first define it as Any to be able to check the that the controller is the correct class

I have removed the calls to Any where it is not needed